ticker.py

```diff
@@ -0,0 +1,135 @@
+from time import time
+import json
+import toml
+import requests
+import argparse
+from urllib import urlencode
+
+PAGE_LIMIT = 200  # for aggregation endpoint
+
+
+def millis():
+    """current time in millis since epoch"""
+    return int(round(time() * 1000))
+
+
+def get_asset_params(prefix, atype, code, issuer):
+    """get aggregation request parameters for single asset"""
+    return {
+        prefix + "_asset_type": atype,
+        prefix + "_asset_code": code,
+        prefix + "_asset_issuer": issuer,
+    }
+
+
+def get_asset_param_from_pair(pair, prefix):
+    """get aggregation parameters for asset pair"""
+    if pair[prefix + "_asset_issuer"] == "native":
+        return get_asset_params(prefix, "native", "", "")
+    else:
+        return get_asset_params(prefix, "credit_alphanum4", pair[prefix + "_asset_code"],
+                                pair[prefix + "_asset_issuer"])
+
+
+def get_aggregation_params(pair, start, end, resolution):
+    """get aggregation request params"""
+    params = {
+        "order": "asc",
+        "limit": PAGE_LIMIT,
+        "start_time": start,
+        "end_time": end,
+        "resolution": resolution
+    }
+    params.update(get_asset_param_from_pair(pair, "base"))
+    params.update(get_asset_param_from_pair(pair, "counter"))
+    return params
+
+
+def sum_tuples(t1, t2):
+    """sum all items in two tuples to a third one. tuples must match in size"""
+    return tuple(sum(t) for t in zip(t1, t2))
+
+
+def record_to_tuple(record):
+    """convert aggregation record to (base_volume, counter_volume, trade_count) tuple"""
+    return float(record["base_volume"]), float(record["counter_volume"]), int(record["trade_count"])
+
+
+def aggregate_pair(horizon_host, pair, start, end, resolution):
+    """
+    fetch all trades from given time period and aggregate
+    :return a tuple of (base_volume, counter_volume, trade_count)
+    """
+    print "aggregating pair:", pair["name"]
+    values = (0, 0, 0)
+    params = get_aggregation_params(pair, start, end, resolution)
+    url = horizon_host + "/trade_aggregations?" + urlencode(params)
+    consumed = False
+    while not consumed:
```

Comment:
potential bug: if the page does not return any results than this may keep spinning in an infinite loop because consumed will never be true?
@tomerweller

Reply:
`records` shows up as an empty array so it stops.
However, will add error handling.

Reply:
yeah makes sense, I misinterpreted that